airbyte-integrations/connectors/source-file/source_file/spec.json

```diff
@@ -0,0 +1,140 @@
+{
+  "documentationUrl": "https://docs.airbyte.io/integrations/sources/file",
+
+  "connectionSpecification": {
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "File Source Spec",
+    "type": "object",
+    "required": ["url", "storage"],
+
+    "properties": {
+      "format": {
+        "type": "string",
+        "enum": [
+          "csv",
+          "json",
+          "html",
+          "excel",
+          "feather",
+          "parquet",
+          "orc",
+          "pickle"
+        ],
+        "default": "csv"
+      },
+      "reader_options": {
+        "type": "string"
+      },
+
+      "storage": {
+        "type": "string",
+        "enum": [
+          "https://",
+          "gs://",
+          "s3://",
+          "scp://",
+          "sftp://",
+          "webhdfs://",
+          "file://"
+        ],
+        "default": "https://"
+      },
+
+      "url": {
+        "type": "string"
+      }
+    },
+
+    "dependencies": {
+      "storage": {
+        "oneOf": [
+          {
+            "properties": {
+              "storage": {
+                "enum": ["gs://"]
+              },
+              "service_account_json": {
+                "type": "string"
+              },
+              "reader_impl": {
```

Comment:
Why do we need that?

Reply:
There are actually 2 ways of accessing both GCS and AWS APIs with this connector:
1. Using the `smart_open` library which gives opportunities to access multiple back-ends too. Unfortunately I've seen  feedback from some users worrying about performances and reliability (maybe due to a wider scope and younger project?)
1. Using more specialized libraries called `gsfs` and `s3fs` which are a bit harder to use but might be slightly faster to transfer files.

I haven't ran any benchmarks yet of course but since we are able to support both APIs, users would have freedom to switch back and forth if needed and consider the impacts of doing so...